### ISSUE TITLE
Pilight nightly crashes Pimatic

### ISSUE THREAD

**Issue by sandervandegeijn** [pimatic/pimatic#389]
There is a problem with the most recent trunk build of pilight (today) and pimatic. The moment I start pilight pimatic discovers it, then pimatic goes nuts. cpu usage is really high, memory usage doubles easilly. Only solution is to kill pilight and then restart pimatic.

**Comment by n3roGit** [comment]
I think sooner or later Pimatic should also be compatible with the new pilight. I'm testing the new raspberry pi 2. Pilight work properly only in the latest nightly... 

At the moment iam running pilight on my old pi1 and on the new p2 everything else and pimatic.

**Comment by Yves911** [comment]
@n3roGit maybe you can give a try to http://pimatic.org/plugins/pimatic-homeduino/ it works very well and support more an more protocols...

**Comment by leader21** [comment]
> I think sooner or later Pimatic should also be compatible with the new pilight

well that depends if there's somebody who will develop the pilight plugin into that direction.
pilight changed its API. so the plugin needs to be updated to work again.
pimatic is able to work without pilight, the RF functionality can be achieved with homeduino. 
ok, pilight has some more protocols available. but if anything important is missing at pimatic, i'm sure sooner or later it will be implemented. 

if every user would spend 20 euro to sweetpi, he could do 4 weeks of holiday in a 5 star resort and develop only pimatic and nothing else. that would be fantastic, don't you think :joy:

if anybody is able to change the plugin feel free to do so. sweetpi has actually some more important projects running with pimatic development and time is short.
 
**the actual state of pilight is : not supported anymore (unless anybody changes the plugin ;)**

**Comment by sweetpi** [comment]
> if every user would spend 20 euro to sweetpi, he could do 4 weeks of holiday in a 5 star resort and develop only pimatic and nothing else. that would be fantastic, don't you think

Such holiday would be great ;) But I'm developing pimatic because it's fun and I love to see how it is used and how people contribute to it. This means that its a free time project for me and I can't work 24/7 on it to implement things people want. But I'm doing my best to get the cool things done :).

As pilight wants to become a full home automation solution it makes less and less sense to use it in pimatic as an plugin, because of the big overlaps. I think pilight is a great solution for 433mhz rf, but to heavy to use it with pimatic in a long term. Further we have an lightweight, flexible and powerful alternative with pimatic-homeduino. So in a long term you have decide between pimatic or pilight.

If someone wants to maintain the pimatic-pilight plugin it would be great and I surely will support it, but as leader21 said, my priorities are different.

**Comment by n3roGit** [comment]
Ah okay. I dont know that pilight isnt supported anymore. Now i will buy everything for the homeduino. 

I just need the Arduino Nano. anything else I can use from before, right? 

And I also thank you very much that you're doing this project forward. I use it very much and it works just great.

**Comment by leader21** [comment]
> I just need the Arduino Nano. anything else I can use from before, right?

yes! you can still use your rf receiver and transmitter. if you're having a breadboard, you can even put the arduino nano on it and wire your devices up. it's easy to install and easy to use, you will see. if it once is running, you will not have anything else but pimatic and homeduino anymore ;)

@sweetpi 
> Such holiday would be great ;) But I'm developing pimatic because it's fun

oh I would say this is lots of fun there :smile: lying at the poolchair with your notebook on your lap, to the left the cocktail, to the right a beautiful woman and up a blue sky. well,if I had the possibility - i would go for the woman :joy: :joy:

**Comment by neographikal** [comment]
And I can imagine, although I truly respect en appreciate the effort Curly put in I do understand it is not ideal. With the homeduino you essentially replaced Pilight as a RF solution and I love how you created this solution in Node :)

One question though, if I switch to homeduino, is the unit/id combination the same as in Pimatic. If not, I'd need to remove a few KAKU switches behind wall outlets which I rather would not do. Also, I already have a Arduino Mega, I suppose I can use that as well?

Thanks!

**Comment by leader21** [comment]
> that one is still in China btw

yeah, it is, just to show the correct arduino and the next sentence says to have a look at your own country if its urgent. there are arduino mini without usb port. would be a loss of time if buying these from ebay china.

> remove a few KAKU switches behind wall outlets

OMG!! Don't do that, you won't need to remove anything :smile: The unit and id will change but that's no problem. Kaku, Intertechno, Duewi and some others are using the same protocol.
have a look here : #380 

everything will be fine!

**Comment by neographikal** [comment]
Sorry, I have some newkaku stuff which learns id/group by pressing the on switch after inserting it into the wall or by pressing a button. It would really help if I knew the conversion. When I came from Nodo to pilight I figured it out by running the low level pilight stuff in a bash script (just go through all the combinations) :)

**Comment by leader21** [comment]
when rf receiver and transmitter is attached to the arduino and homeduino is installed properly, you can press on or off at your switches and they send their actual unit/id to pimatic and you can read the id at the message output from homeduino debug. with this you can set up your device then.

the other way would be to set up a pimatic device with a guilty unit/id, then set the kaku device into learning mode and turn the device on in pimatic. oimatic will send the unit and id over RF and the switch will save it. this works with my intertechno self learning devices. never had any problems with it.

**Comment by neographikal** [comment]
My problem is this: the devices in the wall have learned the unit/org in the pilight format. I can't relearn them anything without taking the wall outlets apart, something I'd rather not do. To solve this when I migrated from nodo to pilight I wrote a bash script that looped through all org/id combo's untill I heard the switch go on/off. That way I could make a conversion table.

Has anyone done this for pimatic vs pilight id's?

**Comment by leader21** [comment]
not that i know of. and as i said - that's imho not necessary! 
push your wall switch, it will send, pimatic will receive the already stored unit and id and you can create a device.
try it and you will see that i'm right. If not i owe you a :beer: : :smile:  you don't have much to loose but only a few euro and a little time to set it up.

anybody else here to convince neographikal ;)

**Comment by neographikal** [comment]
Hmm is there a simple solution to create a bash/python/whatever script that will loop through all combinations with homeduino? Sorry, I'm a bit of newb when it comes to homeduino :)

**Comment by Icesory** [comment]
It is easier to look in the pilight and pimatic protocol implementations. I think you can convert the id through this way.
